### PR TITLE
cgroups/v1: optimize path usage

### PR DESCRIPTION
**Pull request by kolyshkin**
_this is separated out from https://github.com/opencontainers/runc/pull/2438 in order to make review easier_

Limit the use of `getSubsystemPath()` and `d.path()` functions, reusing the `m.paths` instead.

See individual commit descriptions for details.

Number of times mountinfo was parsed during `runc --systemd-cgroup run` of a busybox container running `sleep 1` went down from 50 to 19 as a result of these changes.

### CODE REVIEW

**Comment by mrunalp** [comment]
Needs rebase.

**Comment by kolyshkin** [comment]
rebased

**Comment by kolyshkin** [comment]
@AkihiroSuda PTAL

File: libcontainer/cgroups/fs/fs.go [hunk 1]
```diff
@@ -220,26 +220,26 @@ func (m *manager) Apply(pid int) (err error) {
...
-	d, err := getCgroupData(m.cgroups, pid)
-	if err != nil {
-		return err
-	}
-
 	m.paths = make(map[string]string)
 	if c.Paths != nil {
 		for name, path := range c.Paths {
-			_, err := d.path(name)
+			cgMap, err := cgroups.ParseCgroupFile("/proc/self/cgroup")
```
Comment (kolyshkin) [inline libcontainer/cgroups/fs/fs.go:226]: This obviously need to be taken out of the `for ... range c.Paths` loop!

Comment (kolyshkin) [reply libcontainer/cgroups/fs/fs.go:226]: otherwise we parse /proc/self/cgroup a few times which is absolutely not needed

Comment (kolyshkin) [reply libcontainer/cgroups/fs/fs.go:226]: updated

File: libcontainer/cgroups/fs/fs.go [hunk 2]
```diff
@@ -220,26 +220,26 @@ func (m *manager) Apply(pid int) (err error) {
...
+			return err
+		}
 		for name, path := range c.Paths {
-			_, err := d.path(name)
-			if err != nil {
-				if cgroups.IsNotFound(err) {
-					continue
-				}
-				return err
+			// XXX(kolyshkin@): why this check is needed?
```
Comment (AkihiroSuda) [inline libcontainer/cgroups/fs/fs.go:230]: What happens without this check?

Comment (kolyshkin) [reply libcontainer/cgroups/fs/fs.go:230]: As far as I can see, nothing bad. The `paths` map is then used to write pid to cgroups, and `WriteCgroupProc` has a check that path exists.

There may be other scenarios (with software that uses runc/libcontainer) I don't envision, and I have broken stuff in the past, so I am trying to be extra careful here.

So, ideally, I'd remove this, but maybe later, thus `TODO`.

The new check is already 100x better than the old one, so this patch is a big improvement -- I'm just not sure if the check is needed at all.

**Comment by kolyshkin** [comment]
@AkihiroSuda PTAL

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**